crates/common/src/unit_tests/mod.rs

```diff
@@ -0,0 +1,470 @@
+use alloy::primitives::{Address, U256};
+use alloy_ethers_typecast::transaction::{ReadableClientError, ReadableClientHttp};
+use dotrain::{error::ComposeError, RainDocument, Rebind};
+use futures::TryFutureExt;
+use proptest::{
+    prelude::RngCore,
+    test_runner::{RngAlgorithm, TestRng},
+};
+use rain_interpreter_bindings::IInterpreterStoreV1::FullyQualifiedNamespace;
+use rain_interpreter_eval::{
+    error::ForkCallError,
+    eval::ForkEvalArgs,
+    fork::{Forker, NewForkedEvm},
+    trace::{RainEvalResultError, RainEvalResults},
+};
+use rain_orderbook_app_settings::{
+    blocks::BlockError, config::*, deployer::Deployer, network::Network, unit_test::TestConfig,
+};
+use std::sync::Arc;
+use thiserror::Error;
+use url::Url;
+
+#[derive(Clone)]
+pub struct TestRunner {
+    pub forker: Forker,
+    pub dotrains: Dotrains,
+    pub settings: Settings,
+    pub rng: TestRng,
+    pub test_setup: TestSetup,
+}
+
+#[derive(Clone)]
+pub struct TestSetup {
+    pub block_number: u64,
+    pub deployer: Arc<Deployer>,
+    pub scenario_name: String,
+}
+
+#[derive(Clone)]
+pub struct Dotrains {
+    pub main_dotrain: String,
+    pub test_dotrain: String,
+}
+
+#[derive(Clone)]
+pub struct Settings {
+    pub main_config: Config,
+    pub test_config: TestConfig,
+}
+
+#[derive(Error, Debug)]
+pub enum TestRunnerError {
+    #[error("Scenario not found")]
+    ScenarioNotFound(String),
+    #[error(transparent)]
+    ReadableClientHttpError(#[from] ReadableClientError),
+    #[error(transparent)]
+    BlockError(#[from] BlockError),
+    #[error(transparent)]
+    ForkCallError(#[from] ForkCallError),
+    #[error(transparent)]
+    JoinError(#[from] tokio::task::JoinError),
+    #[error(transparent)]
+    ComposeError(#[from] ComposeError),
+    #[error(transparent)]
+    RainEvalResultError(#[from] RainEvalResultError),
+}
+
+impl TestRunner {
+    pub async fn new(
+        dotrain: &str,
+        test_dotrain: &str,
+        settings: Config,
+        test_settings: TestConfig,
+        seed: Option<[u8; 32]>,
+    ) -> Self {
+        Self {
+            forker: Forker::new(),
+            dotrains: Dotrains {
+                main_dotrain: dotrain.into(),
+                test_dotrain: test_dotrain.into(),
+            },
+            settings: Settings {
+                main_config: settings,
+                test_config: test_settings,
+            },
+            rng: TestRng::from_seed(RngAlgorithm::ChaCha, &seed.unwrap_or([0; 32])),
+            test_setup: TestSetup {
+                block_number: 0,
+                deployer: Arc::new(Deployer {
+                    address: Address::default(),
+                    network: Arc::new(Network {
```

Comment:
if this is supposed to be some kind of dummy network, can it be an impl on `Network`? this logic looks like bad boilerplate here